CorsixTH/Lua/entities/humanoids/staff/doctor.lua

```diff
@@ -182,14 +161,125 @@ function Doctor:updateSkill(consultant, trait, amount) -- luacheck: ignore 212
+    local base, extra = 0.9, 0.2 -- specify the +-10% variability
+    local factor = value * extra
+    return base + factor
+  end
+
+  local performers = {
+    doctor_focus = clamp(doc_atd),
+    teacher_ability = clamp(con_skill^2), -- ~1.0 modifier at 0.75 skill
+    random = math.t_random(0.9, 1, 1.1), -- Allow some randomness
+  }
+  -- Average the three performance modifiers to generate a single performance factor
+  local count, p_impact = 0, 0
+  for _, perf in pairs(performers) do
+    p_impact = p_impact + perf
+    count = count + 1
+  end
+  p_impact = p_impact / count
+
+  -- Class size impacts performance when there are more than 3 students at rate 2/n-1
+  local c_impact = class_size >= 3 and class_size - 1 or 2
+  c_impact = 2 / c_impact
+
+  return p_impact * c_impact
+end
+
+--! Calculates the amount of skill to learn towards becoming a doctor/consultant and
+--! any amount towards a specialist skill being taught
+--! There are detailed explanations on the logic to these calculations in the Wiki (see
+--! Training) - PR NOTE: If merge, make wiki page
+--!param consultant (entity) The person teaching the student
+--!param room_factor (number) What score training objects in the room give
+--!param student_count (number) Total number of students present
+function Doctor:trainSkills(consultant, room_factor, student_count)
+  --[[Information about these calculations based on TH:
+    General skill is based on gbv.trainingrate only; specialisms are based on the
+    impact of objects in the room (total training value of relevant objects).
+    Both calculations follow the general formula: x * skills_factor * class_performance
+    where x relates to the gbv.trainingrate (skill) or training value of room (specialism),
+    skills_factor is the impact from number of skills being taught,
+    and class_performance is performance of teacher/student and the class size.
+    For proper control in CorsixTH, specialsms are affected by the AbilityThreshold
+    for the specialism (varies by specialty)
+    The goal is to gain skill to become a consultant (ConsultantTreshold (usually
+    0.750)); or to be qualified in a specialism (always a value of 1.000)
+    PR NOTE: If merge, move to Wiki
+  ]]--
+  local level_config = self.world.map.level_config
+  local general_factor = level_config.gbv.TrainingRate
+  --[[ Scalers and floors:
+    For balance at the very worst it should take ~1000 days to go from skill 0 to skill
+    1. This will be known as the mimum learning value.
+    Generalist skill (g) scales differ from specialism (s) scale to give CorsixTH
+    values. g_scale is a simple conversion of value; but s_scale is more complex as it
+    provides a balance on setting the max skill a doctor can have and still become a
+    surgeon: this is ~0.5 skill (on normal difficulty in base game)
+    training.
+    PR NOTE: If merge, move to Wiki
+  ]]--
+  local MIN_LEARN_VALUE, g_scale, s_scale = 0.001, 10000, 12
```

Comment:
Why not split it into 3 assignments? Easier to read.

Suggestion: You can copy the above text to the wiki but leave a copy here?
At least you have more chance that someone will update the explanation then?

Reply:
Yeah I was just trying to save space. Will revert back to individual lines.